UI/UISearchBar.cs

```diff
@@ -141,6 +150,48 @@ private void HandleTextInput()
 			}
 		}
 
+		private void HandleRight()
+		{
+			if (cursorPosition != Text.Length)
+			{
+				//Check if first character is Whitespace and skip if so
+				if (char.IsWhiteSpace(Text[cursorPosition]))
+				{
+					cursorPosition += 1;
+				}
+
+				var newPos = Text.IndexOf(' ', cursorPosition);
```

Comment:
Does this behave good when the last cursor is at the end of the line?
Does it behave right when the last character is a space and the cursor is one space before last?